### PR TITLE
refactoring!(state): extend SubmitPayForBlob method

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Resolves https://github.com/celestiaorg/celestia-node/issues/1840

## Checklist
- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: state/core_access.go
Comment (Wondertan) [inline state/core_access.go:176]: I think we should just pass the Blob type in the method. Mainly because we have to allow our users to specify share versions and extending one with one more param is a no go

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
* Those who use us must import the app whether we rely on the type or not.
* We cannot make `state` dependant on `blob` as `blob` is supposed to depend on it.
	* However, I assumed the `modblob`, and if we can make the root level `blob` with blob definition, we might avoid cyclical dependency
	* Yeah, `state` would depend on `blob `, and `modblob` would depend on it. No cyclic dep between `modblob` and `state`, so I think we should do it.
	* We can merge this one, then introduce `blob` with the same blob and update this one or punt this PR until we make `blob`, up to @vgonkivs

**Review by renaynay — CHANGES_REQUESTED** [review]
I would actually prefer we hold off on breaking API right now (at least before next release).

**Comment by vgonkivs** [comment]
Created a feature branch for the Blob Module. So, I can merge everything there and then we can test it

**Comment by vgonkivs** [comment]
> I'm worried about adding app types to the API

I agree here with you. We've discussed with Evan extracting the Blob type to a separate repo, so, we can re-use it. But it's not a high priority rn. I'd suggest leaving app types for now, keeping in mind that they will be changed in the future.

@distractedm1nd

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

File: state/core_access.go
Comment (walldiss) [inline state/core_access.go:168]: Did you mean to use b later to protect `blobs` from modifications? There is something wring, since b is not used after copy.

**Review by walldiss — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**
